### DISCUSSION TITLE
Display Response Schmeas similar to Redoc

### DISCUSSION THREAD

**Discussion by Schnorts** [scalar/scalar#1268] [Ideas]
Redoc:
<img width="1490" alt="image" src="https://github.com/scalar/scalar/assets/28060666/306edcca-72c8-47ca-a7d8-7c5f43fd637c">

vs. Scalar:
<img width="1711" alt="image" src="https://github.com/scalar/scalar/assets/28060666/0be6ca85-0290-4cf9-ace4-8eb4c94cb4f4">

**Comment by marclave** [comment]
@Schnorts would you want more information in the `show schema` toggle in the response?

**Reply by Schnorts** [reply]
Hey Marc,
The point is, in Scalar, the schema is displayed as JSON and is not expandable, such as a tree, in the example given in the images, you see, that in the 200 response section of Redoc, you have a potentially infinitely expandable model, `elements` defined as reference to the Challenge schema are correctly displayed and you can clic, yourself through. In Scalar, the JSON ends with elements and you cannot see how the response is structured. This is an issue here.
By the way, i like Scalar for its speed, great achievement! I would advise, when you implement the expandable response definitions, that you render them on demand, since these can potentially be resource hungry and are required only if viewed or eventually on expansion of the 200 response.
Keep on the great work!

**Comment by mattoni** [comment]
Yeah being able to see the TYPE of the response, and the descriptions etc of each would be incredibly valuable. As is, trying to identify the response types means parsing the json, with no descriptions of the fields.

**Comment by marclave** [comment]
this makes sense! okay let me make this an issue : )

**Reply by lephuongbg** [reply]
I think the "Show Schema" should use the design of "Models" section (which show attributes of the model, along with description for each attribute), instead of showing the raw JSON schema, which is unhelpful as a documentation (but still helpful if user wants to copy the schema and do something with it in their code).

![image](https://github.com/user-attachments/assets/e0e73964-323b-4d01-90f4-e73504c1cb82)
